crates/core/tedge_mapper/src/c8y/service_monitor.rs

```diff
@@ -50,36 +46,44 @@ impl TopicInfo {
     }
 }
 
-pub fn convert_health_status_message(message: &Message, device_name: String) -> Vec<Message> {
+pub fn convert_health_status_message(
+    message: &Message,
+    device_name: String,
+    default_service_type: String,
+) -> Vec<Message> {
     let mut mqtt_messages: Vec<Message> = Vec::new();
     let topic = message.topic.name.to_owned();
     let topic_info = TopicInfo::parse_topic_info(&topic);
+    let default_health_status = format!("\"type\":{default_service_type},\"status\":\"unknown\"");
 
     // If not Bridge health status
     if !topic_info.service_name.contains("bridge") {
-        let payload_str = message
-            .payload_str()
-            .unwrap_or(r#""type":"service","status":"unknown""#);
+        let payload_str = message.payload_str().unwrap_or(&default_health_status);
 
         let mut health_status =
             serde_json::from_str(payload_str).unwrap_or_else(|_| HealthStatus {
-                service_type: "service".to_string(),
+                service_type: Some(default_service_type.clone()),
                 status: "unknown".to_string(),
             });
 
         if health_status.status.is_empty() {
             health_status.status = "unknown".into();
         }
 
-        if health_status.service_type.is_empty() {
-            health_status.service_type = "service".into();
+        match health_status.service_type {
+            Some(s_type) if s_type.is_empty() => {
+                health_status.service_type = Some(default_service_type)
+            }
+            None => health_status.service_type = Some(default_service_type),
+            Some(_) => {}
         }
 
         let status_message = service_monitor_status_message(
             &device_name,
             &topic_info.service_name,
             &health_status.status,
-            &health_status.service_type,
+            // Unwrap is safe here because the service_type is always having some value
+            &health_status.service_type.unwrap(),
```

Comment:
Here again I feel weird. `service.type` must have a value, but since it is `Option`, you need to unwrap.